Auto reconnect feature added as an option

andrasbiro opened the pull request:
I've added auto reconnect capabilites to cutecom, similar to what teraterm has. This is useful for USB serial converters, e.g. if you have no other method of resetting a device then unplugging the USB.
Demo:
https://www.dropbox.com/s/2puh3pgmfm7loxd/cutecom_reconnect.mp4?dl=0

cyc1ingsir commented on mainwindow.cpp:
```diff
@@ -420,15 +431,25 @@ void MainWindow::handleError(QSerialPort::SerialPortError error)
+            }
+            m_reconnectTimer.start();
+        } else {
+            QString heading = (m_deviceState == DEVICE_OPENING) ? tr("Error opening device") : tr("Device Error");
```
The comment about the hot unplug and the error reported only once should be moved to here

cyc1ingsir commented on controlpanel.cpp:
```diff
@@ -50,11 +50,14 @@ ControlPanel::ControlPanel(QWidget *parent, Settings *settings)
     connect(m_check_timestamp, &QCheckBox::toggled,
             [=](bool checked) { emit settingChanged(Settings::ShowTimestamp, checked); });
+    connect(m_check_reconnect, &QCheckBox::toggled,
+            [=](bool checked) { emit settingChanged(Settings::AutoReconnect, checked); });
```
Without having tested it. I wonder if it would be useful to call 
`m_reconnectTimer.stop();` here too in case checked is set to false.
Otherwise the user would see an error if CuteCom is still trying to reconnect with no device connected but the checkbox being unchecked, wouldn't he?

andrasbiro commented:
I fixed both. Also added myself to the contributors list :) (for this and #57)

FYI, I have a couple of unrelated feature requests, I will open them once this is closed.

andrasbiro commented:
I realized I used the wrong email address for the commits, so I will recreate this later, possibly after the move mentioned #58
